### PR TITLE
Adding RestActions support for Start/Stop Detector API

### PR DESCRIPTION
**Pull request by saratvemulapalli**
*Issue #195 *

*Description of changes:*
Adding RestActions support for Start/Stop Detector API which is needed to support AD FGAC

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyDetectorJobAction.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.ad.transport;
+
+import org.elasticsearch.action.ActionType;
+
+public class AnomalyDetectorJobAction extends ActionType<AnomalyDetectorJobResponse> {
+    public static final AnomalyDetectorJobAction INSTANCE = new AnomalyDetectorJobAction();
+    public static final String NAME = "cluster:admin/opendistro/ad/detector/startstop";
```
Comment (saratvemulapalli) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyDetectorJobAction.java:22]: Couldn't think of a better name. Please suggest if you have any better idea's/

Comment (ylwu-amzn) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyDetectorJobAction.java:22]: How about change "startstop" to "jobmanagement"

Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyDetectorJobAction.java:22]: Thats a nice idea.

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorJobActionHandler.java [hunk 2]
```diff
@@ -72,42 +74,45 @@
...
+        this.xContentRegistry = xContentRegistry;
     }
 
     /**
      * Start anomaly detector job.
      * 1.If job not exists, create new job.
      * 2.If job exists: a). if job enabled, return error message; b). if job disabled, enable job.
      *
-     * @throws IOException IOException from {@link AnomalyDetectionIndices#getAnomalyDetectorJobMappings}
+     * @throws IOException IOException from //TODO vemsarat@ fix this comment
```
Comment (ohltyler) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorJobActionHandler.java:109]: small: update the link here?

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorJobActionHandler.java [hunk 3]
```diff
@@ -120,34 +125,40 @@ private void onCreateMappingsResponse(CreateIndexResponse response) throws IOExc
             prepareAnomalyDetectorJobIndexing();
         } else {
             logger.warn("Created {} with mappings call not acknowledged.", ANOMALY_DETECTORS_INDEX);
-            channel
-                .sendResponse(
-                    new BytesRestResponse(RestStatus.INTERNAL_SERVER_ERROR, response.toXContent(channel.newErrorBuilder(), EMPTY_PARAMS))
+            listener
+                .onFailure(
+                    new ElasticsearchStatusException(
+                        "Created " + ANOMALY_DETECTORS_INDEX + "with mappings call not acknowledged.",
```
Comment (ohltyler) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorJobActionHandler.java:131]: small: add space before `with` here

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyDetectorJobAction.java (hunk 1 above)
Comment (ohltyler) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyDetectorJobAction.java:22]: +1 to "jobmanagement", or even just "job"/"jobs"

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyDetectorJobTransportAction.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        );
+
+        if (rawPath.endsWith(RestHandlerUtils.START_JOB)) {
+            try {
+                handler.startAnomalyDetectorJob();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else if (rawPath.endsWith(RestHandlerUtils.STOP_JOB)) {
+            handler.stopAnomalyDetectorJob(detectorId);
```
Comment (ohltyler) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyDetectorJobTransportAction.java:86]: encapsulate this in a `try/catch` as well?

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyDetectorJobAction.java (hunk 1 above)
Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyDetectorJobAction.java:22]: Sure i'll go with jobmanagement.

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorJobActionHandler.java (hunk 2 above)
Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorJobActionHandler.java:109]: Sure nice catch will fix it.

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorJobActionHandler.java (hunk 3 above)
Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorJobActionHandler.java:131]: Sure..

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyDetectorJobTransportAction.java (hunk 4 above)
Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyDetectorJobTransportAction.java:86]: Sure..

**Review by ohltyler — APPROVED** [review]
LGTM!

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java [hunk 5]
```diff
@@ -171,6 +171,7 @@ private void prepareAnomalyDetectorIndexing() {
         if (method == RestRequest.Method.PUT) {
             handler.getDetectorJob(clusterService, client, detectorId, listener, () -> updateAnomalyDetector(detectorId), xContentRegistry);
         } else {
+            logger.info("#177 creating Anomaly Detector");
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java:174]: Clean debugging log?

Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java:174]: Sure. Probably something messed up during the merge.

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

**Pull request merged**